libraries/shared/factories/event/repository_test.go

```diff
@@ -0,0 +1,211 @@
+			Expect(res.Variable1).To(Equal(testModel.ColumnValues["variable1"]))
+		})
+
+		Describe("returns errors", func() {
+			It("for empty model slice", func() {
+				err := event.Create([]event.InsertionModel{}, db)
+				Expect(err).To(MatchError("repository got empty model slice"))
+			})
+
+			It("for failed SQL inserts", func() {
+				header := fakes.GetFakeHeader(1)
+				headerID, headerErr := headerRepository.CreateOrUpdateHeader(header)
+				Expect(headerErr).NotTo(HaveOccurred())
+
+				brokenModel := event.InsertionModel{
+					SchemaName: "public",
+					TableName:  "testEvent",
+					// Wrong name of last column compared to DB, will generate incorrect query
+					OrderedColumns: []event.ColumnName{
+						event.HeaderFK, event.LogFK, "variable2",
+					},
+					ColumnValues: event.ColumnValues{
+						event.HeaderFK: headerID,
+						event.LogFK:    logID,
+						"variable1":    "value1",
+					},
+				}
+
+				// Remove cached queries, or we won't generate a new (incorrect) one
+				delete(event.ModelToQuery, "publictestEvent")
+
+				createErr := event.Create([]event.InsertionModel{brokenModel}, db)
+				Expect(createErr).To(HaveOccurred())
+
+				// Remove incorrect query, so other tests won't get it
+				delete(event.ModelToQuery, "publictestEvent")
+			})
+
+			It("for unsupported types in ColumnValue", func() {
+				testModel = event.InsertionModel{
+					SchemaName: "public",
+					TableName:  "testEvent",
+					OrderedColumns: []event.ColumnName{
+						event.HeaderFK, event.LogFK, "variable1",
+					},
+					ColumnValues: event.ColumnValues{
+						event.HeaderFK: headerID,
+						event.LogFK:    logID,
+						"variable1":    big.NewInt(5), // Not a supported builtin type
+					},
+				}
+
+				createErr := event.Create([]event.InsertionModel{testModel}, db)
+				Expect(createErr).To(MatchError(event.ErrUnsupportedValue))
+			})
+		})
+
+		It("upserts queries with conflicting source", func() {
+			conflictingModel := event.InsertionModel{
+				SchemaName: "public",
+				TableName:  "testEvent",
+				OrderedColumns: []event.ColumnName{
+					event.HeaderFK, event.LogFK, "variable1",
+				},
+				ColumnValues: event.ColumnValues{
+					event.HeaderFK: headerID,
+					event.LogFK:    logID,
+					"variable1":    "conflictingValue",
+				},
+			}
+
+			createErr := event.Create([]event.InsertionModel{testModel, conflictingModel}, db)
+			Expect(createErr).NotTo(HaveOccurred())
+
+			var res TestEvent
+			dbErr := db.Get(&res, `SELECT log_id, variable1 FROM public.testEvent;`)
+			Expect(dbErr).NotTo(HaveOccurred())
+			Expect(res.Variable1).To(Equal(conflictingModel.ColumnValues["variable1"]))
+		})
+
+		It("generates correct queries", func() {
+			actualQuery := event.GenerateInsertionQuery(testModel)
+			expectedQuery := `INSERT INTO public.testEvent (header_id, log_id, variable1) VALUES($1, $2, $3)
+		ON CONFLICT (header_id, log_id) DO UPDATE SET header_id = $1, log_id = $2, variable1 = $3;`
```

Comment:
Really minor but we can probably skip updating the header_id and log_id if we hit this conflict, right? Are we including that just because it's easier to automate?

Reply:
@rmulhol Elaborate? I just made this return whatever was written manually in the transformer repositories before.